### PR TITLE
force a valid backlight on setting on color lcd radios

### PR DESCRIPTION
**Pull request by gagarinlg**
Summary of changes:
When configuration is broken it can happen, that on Color LCD radios the backlight is set to off and/or the brightness is set to off and/or the backlight timer is set to 0.
All of those conditions render the radios unusable.
These changes enforce usable backlight settings, even when they are overwritten by functions or lua scripts.

### CODE REVIEW

File: radio/src/opentx.cpp [hunk 1]
```diff
@@ -1879,6 +1888,10 @@ void opentxInit()
     // no backlight mode off on color lcd radios
     g_eeGeneral.backlightMode = e_backlight_mode_keys;
   }
+  if (g_eeGeneral.backlightBright <= BACKLIGHT_LEVEL_MIN)
+    g_eeGeneral.backlightBright = BACKLIGHT_LEVEL_MIN+5;
```
Comment (raphaelcoeffic) [inline radio/src/opentx.cpp:1892]: What is actually the minimum here? Is that `BACKLIGHT_LEVEL_MIN`, `BACKLIGHT_LEVEL_MIN + 1`, or `BACKLIGHT_LEVEL_MIN + 5`?

We could just as well set `BACKLIGHT_LEVEL_MIN` to the value corresponding to the current `BACKLIGHT_LEVEL_MIN + 5` and just enforce relative to `BACKLIGHT_LEVEL_MIN`.

Comment (gagarinlg) [reply radio/src/opentx.cpp:1892]: > What is actually the minimum here? Is that `BACKLIGHT_LEVEL_MIN`, `BACKLIGHT_LEVEL_MIN + 1`, or `BACKLIGHT_LEVEL_MIN + 5`?

The idea here is to allow the user to set the backlight to `BACKLIGHT_LEVEL_MIN + 1` on purpose, but when something goes wrong and it is set to off, you get, as a failsave, a level where you can read something on the display when you are outside.



> We could just as well set `BACKLIGHT_LEVEL_MIN` to the value corresponding to the current `BACKLIGHT_LEVEL_MIN + 5` and just enforce relative to `BACKLIGHT_LEVEL_MIN`.

This does not work, as `BACKLIGHT_LEVEL_MIN` is also the lower bound fot the backlight off brightness, or do you mean we always add 5 to the on brightness and limit it to `BACKLIGHT_LEVEL_MAX`?

Comment (gagarinlg) [reply radio/src/opentx.cpp]: What I did not take into account is the inverted baclight level usage on color lcd radios. I pushed an update that fixes this and also removed to offsets. You are right about the offsets, it is better to just increase the minimum value, if needed.

thx

File: radio/src/opentx.cpp [hunk 2]
```diff
@@ -568,6 +568,11 @@ void checkBacklight()
       }
       if (backlightOn) {
         currentBacklightBright = requiredBacklightBright;
+#if defined(COLOR_LCD)
+        // force backlight on for color lcd radios
+        if(currentBacklightBright > BACLIGHT_LEVEL_MAX - BACKLIGHT_LEVEL_MIN)
+          currentBacklightBricht = BACKLIGHT_LEVEL_MAX - BACKLIGHT_LEVEL_MIN;
```
Comment (raphaelcoeffic) [inline radio/src/opentx.cpp:574]: `currentBacklightBricht`? Typo?

Comment (gagarinlg) [reply radio/src/opentx.cpp]: Yes and los COLOR_LCD instead of COLORLCD, so the compiler did not find it.

**Comment by raphaelcoeffic** [comment]
Looks good to me now, let's handout to the test department 😅

**Comment by raphaelcoeffic** [comment]
I think we should in fact enforce a minimum for `lightAutoOff` of maybe 5s in the UI as well.

**Review by pfeerick — APPROVED** [review]
Perfect! I deliberately set my backlight to inverted 0, saw the screen blank come back up to min brightness, for throttle warning and then the global function via pot brightness kicked in once that was cleared. Obviously if I hadn't had the global function that would have meant no backlight at all prior to this PR. Nice! :)

**Comment by gagarinlg** [comment]
I think the main reason to have this problem is what I described in  #1412 or some user error, of course.

**Comment by pfeerick** [comment]
Raphael also suspects that Companion 2.5 has a nice default as well ;)

**Pull request merged**